test/lib/commands/adduser.js

```diff
@@ -1,190 +1,176 @@
-  scope: '',
-  fromFlatOptions: true,
-}
-
-let failSave = false
-let deletedConfig = {}
-let registryOutput = ''
-let setConfig = {}
-const authDummy = (npm, options) => {
-  if (!options.fromFlatOptions) {
-    throw new Error('did not pass full flatOptions to auth function')
-  }
-
-  return Promise.resolve({
-    message: 'success',
-    newCreds: {
-      username: 'u',
-      password: 'p',
-      email: 'u@npmjs.org',
-    },
-  })
-}
-
-const deleteMock = (key, where) => {
-  deletedConfig = {
-    ...deletedConfig,
-    [key]: where,
-  }
-}
-const npm = {
-  flatOptions: _flatOptions,
-  config: {
-    delete: deleteMock,
-    get (key, where) {
-      if (!where || where === 'user') {
-        return _flatOptions[key]
-      }
-    },
-    getCredentialsByURI,
-    async save () {
-      if (failSave) {
-        throw new Error('error saving user config')
-      }
-    },
-    set (key, value, where) {
-      setConfig = {
-        ...setConfig,
-        [key]: {
-          value,
-          where,
-        },
-      }
-    },
-    setCredentialsByURI,
-  },
-  output: msg => {
-    result = msg
-  },
-}
-
-const AddUser = t.mock('../../../lib/commands/adduser.js', {
-  npmlog: {
-    clearProgress: () => null,
-    disableProgress: () => null,
-  },
-  'proc-log': {
-    notice: (_, msg) => {
-      registryOutput = msg
-    },
-  },
-  '../../../lib/auth/legacy.js': authDummy,
-})
-
-const adduser = new AddUser(npm)
-
-t.afterEach(() => {
-  _flatOptions.authType = 'legacy'
-  _flatOptions.scope = ''
-  registryOutput = ''
-  deletedConfig = {}
-  setConfig = {}
-  result = ''
-  failSave = false
-})
+const { load: loadMockNpm } = require('../../fixtures/mock-npm.js')
+const mockGlobals = require('../../fixtures/mock-globals.js')
+const MockRegistry = require('../../fixtures/mock-registry.js')
+const stream = require('stream')
 
 t.test('usage', async t => {
+  const { npm } = await loadMockNpm(t)
+  const adduser = await npm.cmd('adduser')
   t.match(adduser.usage, 'adduser', 'usage has command name in it')
 })
 
 t.test('simple login', async t => {
-  await adduser.exec([])
-  t.equal(
-    registryOutput,
-    'Log in on https://registry.npmjs.org/',
-    'should have correct message result'
-  )
-
-  t.same(
-    deletedConfig,
-    {
-      _token: 'user',
-      _password: 'user',
-      username: 'user',
-      _auth: 'user',
-      _authtoken: 'user',
-      '-authtoken': 'user',
-      _authToken: 'user',
-      '//registry.npmjs.org/:-authtoken': 'user',
-      '//registry.npmjs.org/:_authToken': 'user',
-      '//registry.npmjs.org/:_authtoken': 'user',
-      '//registry.npmjs.org/:always-auth': 'user',
-      '//registry.npmjs.org/:email': 'user',
+  const stdin = new stream.PassThrough()
+  stdin.write('test-user\n')
+  stdin.write('test-password\n')
+  stdin.write('test-email@npmjs.org\n')
+  mockGlobals(t, {
+    'process.stdin': stdin,
+    // 'process.stdout': new stream.PassThrough(), // to quiet readline
```

Comment:
this can be removed

Reply:
Needs to be uncommented, this was me debugging